sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -50,24 +50,24 @@ def __init__(self, key):
     @property
     def key(self):
         # type () -> str
-        """The value of the configured API key.
+        """The value of the configured key.
 
         :rtype: str
         """
         return self._key
 
     def update(self, key):
         # type: (str) -> None
-        """Update the API key.
+        """Update the key.
 
-        This can be used when you've regenerated your service API key and want
+        This can be used when you've regenerated your service key and want
         to update long-lived clients.
 
-        :param str key: The API key to your Azure account.
+        :param str key: The key to your Azure account.
         :raises: TypeError
         """
         if not key:
-            raise TypeError("The API key used for updating can not be None or empty")
+            raise TypeError("The key used for updating can not be None or empty")
```

Comment:
This would raise `TypeError` for an empty string. Empty string should presumably be a `ValueError`.